tests/clients/test_experiment.py

```diff
@@ -348,334 +357,315 @@ def test_experiment_run_data_schema_multiple():
     ],
 )
-def test_single_filter_validation(by, key, operator, value, message):
-    with pytest.raises(ValueError, match=message):
-        SingleFilter(by, key, operator, value)
+def test_filter_schema_invalid_operator_with_key(filter_type, value, operator):
+    filter = filter_type("key", operator, value)
+    with pytest.raises(ValidationError, match="Not a discrete operator"):
+        FilterSchema().dump(filter)
 
 
-def test_compound_filter_validation():
+@pytest.mark.parametrize(
+    "operator, expected_operator",
+    [(LogicalOperator.AND, "and"), (LogicalOperator.OR, "or")],
+)
+def test_filter_schema_compound(operator, expected_operator):
+    filter = CompoundFilter(operator, [PROJECT_ID_FILTER, TAG_FILTER])
+    data = FilterSchema().dump(filter)
+    assert data == {
+        "operator": expected_operator,
+        "conditions": [PROJECT_ID_FILTER_BODY, TAG_FILTER_BODY],
+    }
+
+
+def test_filter_schema_nested():
     filter = CompoundFilter(
-        operator=CompoundFilterOperator.OR,
-        conditions=[
-            SingleFilter(
-                SingleFilterBy.TAG,
-                "tag_key",
-                SingleFilterOperator.EQUAL_TO,
-                "tag_value",
+        LogicalOperator.AND,
+        [
+            CompoundFilter(
+                LogicalOperator.AND, [PROJECT_ID_FILTER, TAG_FILTER]
+            ),
+            CompoundFilter(
+                LogicalOperator.OR, [TAG_FILTER, PROJECT_ID_FILTER]
             ),
-            None,
         ],
     )
-    run_query = QueryRuns(filter, None, None)
-    with pytest.raises(ValidationError):
-        QueryRunsSchema().dump(run_query)
+    data = FilterSchema().dump(filter)
+    assert data == {
+        "operator": "and",
+        "conditions": [
+            {
+                "operator": "and",
+                "conditions": [PROJECT_ID_FILTER_BODY, TAG_FILTER_BODY],
+            },
+            {
+                "operator": "or",
+                "conditions": [TAG_FILTER_BODY, PROJECT_ID_FILTER_BODY],
+            },
+        ],
+    }
 
 
 @pytest.mark.parametrize(
-    "by, key, by_body",
+    "sort_type, by",
     [
-        (SortBy.STARTED_AT, None, "startedAt"),
-        (SortBy.RUN_NUMBER, None, "runNumber"),
-        (SortBy.DURATION, None, "duration"),
-        (SortBy.TAG, "tag-key", "tag"),
-        (SortBy.PARAM, "param-key", "param"),
-        (SortBy.METRIC, "metric-key", "metric"),
+        (StartedAtSort, "startedAt"),
+        (RunNumberSort, "runNumber"),
+        (DurationSort, "duration"),
     ],
 )
 @pytest.mark.parametrize(
-    "order, order_body", [(SortOrder.ASC, "asc"), (SortOrder.DESC, "desc")]
+    "order, expected_order", [(SortOrder.ASC, "asc"), (SortOrder.DESC, "desc")]
 )
-def test_sort_schema(by, key, by_body, order, order_body):
-    sort = Sort(by, key, order)
-    expected_json = {"by": by_body, "key": key, "order": order_body}
+def test_sort_schema_no_tag(sort_type, by, order, expected_order):
```

Comment:
Is it better to name it as test_sort_schema_no_key instead of test_sort_schema_no_tag, because we have been using no key in the filter tests, also, the key field is named key on construction of a sort instance.

Reply:
Ah, this was a mistake, good spot.